### ISSUE TITLE
User Profile - Scroll to section you came from when returning from an edit page.

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#4285]
During user testing the user expressed frustration that every time she returned from an edit page she had to scroll back down from the top to find where she left off.  She said she wished the page returned to it's original position every time she clicked "Save and go back" on an edit page.

**Comment by vd1992** [comment]
proposal: 
add routing like: 
```
myProfileSection: (section): string => path.join("/", lang, "talent", `profile#${section}`),
```

to be triggered in this function for all forms 
https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/frontend/talentsearch/src/js/components/aboutMeForm/AboutMeForm.tsx#L93-L98

**Comment by gobyrne** [comment]
Is this also a problem with the back button? Or just using the "return" links?

**Comment by esizer** [comment]
We have this as a feature in #4597 using [`<ScrollRestoration />`](https://reactrouter.com/en/main/components/scroll-restoration) . However, it specifically does not work for "save and go back" because, on the profile it runs a mutation that invalidates our cache and re-runs the query in our authorization provider causing the entire app to re-render. 

If we could somehow make the `urql` cache to only mark this query as stale when the important info changes (login, logout, roles changed) this should work.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/8992d7b8845c9d2315c80e2fea0358288d2f3b3a/frontend/common/src/components/context/AuthorizationProvider.tsx#L8-L10

**Comment by tristan-orourke** [comment]
@esizer I think this is another ticket solved by #4597 😁

**Comment by esizer** [comment]
> @esizer I think this is another ticket solved by https://github.com/GCTC-NTGC/gc-digital-talent/pull/4597 😁

Does it work?!? 😮
